### ISSUE TITLE
development experience improvements

### ISSUE THREAD

**Issue by edenvidal** [rnbwdev/rnbw#545]
- [ ] setup codespaces
https://github.com/features/codespaces
https://github.blog/2023-05-10-revolutionize-your-open-source-workflows-the-top-3-reasons-why-github-codespaces-is-a-must-have-for-maintainers/
- [ ] add bundling/releasing docs to contributing.md
- [ ] make every external resouce an actual package so it would work offline.
- [ ] Have the github-pages build the rnbw.dev from the main branch. so that whenever we merge to the main, it gets there.

code optimizations:
- [ ] import/export of modules
- [ ] optimize re-render of the components by memo, useMemo
- [ ] useCallback for functions
- [ ] more will be added ...

**Comment by edenvidal** [comment]
@atulbhatt-system32 the top list is a priority 👆
alsom are these relevant:
code optimizations (added by @jasonliu7040 )
- [ ] import/export of modules
- [ ] optimize re-render of the components by memo, useMemo
- [ ] useCallback for functions
- [ ] more will be added ...

**Comment by atulbhatt-system32** [comment]
@edenvidal 
- We are already doing the import and export thing correctly.
- We should not do memo and useMemo until we face performance issue. For now there is one issue you created recently regarding "add" being slow. So we will have to see if this can be useful there. But using memo and useMemo is not always required.
- It is similar to memo and useMemo thing. We are already doing it wherever we felt the need for it.

So I will be marking it is not relevant as of now. For optimizations we will focus on individual part of the code which actually require them.

**Comment by atulbhatt-system32** [comment]
@edenvidal regarding making every external resource an actual package - wdym by running them offline?

If the users are going to use rnbw from rnbw.dev then in both situations - current and with external resource as package they will be required to have the internet connection.

If you mean that when running locally, just how we run currently by doing `npm start` they only do `npm install` once and never worry about connecting with internet then it makes sense.

I think we can pick up making external dependencies as packages later after we fix other bugs. Because we would also need to add type support to them.

Would love to know your take on this!

**Comment by edenvidal** [comment]
@atulbhatt-system32 yes. i mean that when running locally, you should be able to do npm install and that's it. no online dependencies. in order to do that (and for more excellent benefits), we'll have to make all of our own repos to npm packages

**Comment by atulbhatt-system32** [comment]
Sure understood at @edenvidal. I strongly advocate on picking it later and also get autocomplete feature by adding type support. This will ease the use of rene, svg-icon so that we don't have to remember all the classes or icon names.

**Comment by edenvidal** [comment]
@atulbhatt-system32 sounds good. Maybe this can be @stanislavasal task, as it'll encourage her to see the project holistically.

**Comment by atulbhatt-system32** [comment]
sounds good @edenvidal. As I remember my first task with you was also similar making rene an npm package.

**Comment by edenvidal** [comment]
@atulbhatt-system32 move it back to "in progress"

**Comment by atulbhatt-system32** [comment]
@edenvidal I removed it from in progress because the ticket inside it is already in progress. If this is creating confusion so let's have it also in `in progress`

The package details will all be update in docs once this master this issue we have closed entirely.
